### ISSUE TITLE
Allow storage classes to be designated as the default class

### ISSUE THREAD

**Issue by sdavids13** [terraform-providers/terraform-provider-kubernetes#173]
### Affected Resource(s)
Please list the resources as a list, for example:
- kubernetes_storage_class

### Terraform Configuration Files
```hcl
resource "kubernetes_storage_class" "standard" {
  metadata {
    name = "${local.standard_storage_class_name}"
    annotations = {
      "storageclass.kubernetes.io/is-default-class" = "true"
    }
  }
  storage_provisioner = "kubernetes.io/aws-ebs"
  parameters {
    type      = "gp2"
    encrypted = "true"
  }
}
```
### Expected Behavior
Created the "standard" storage class and made it the system default storage class.

### Actual Behavior
Terraform refuses to run saying: `Error: module.eks.kubernetes_storage_class.standard: metadata.0.annotations: "storageclass.kubernetes.io/is-default-class" is internal Kubernetes annotation`

### References
There was an active decision made to not allow these "internal" annotation found in #50. But, as discussed in #60, this is an annotation documented in the [Kubernetes documentation](https://kubernetes.io/docs/tasks/administer-cluster/change-default-storage-class/) that is valid for end-users to update themselves. There should be some sort of whitelist or `force` flag to go ahead and move forward with some of these valid (not internal only) annotations.

**Comment by rfvermut** [comment]
Hi! Any plans on this one? Or workarounds?

**Comment by Artemu** [comment]
> Hi! Any plans on this one? Or workarounds?

This is how I work around it:

```
resource "null_resource" "storge_patch" {
  provisioner "local-exec" {
    command = "kubectl patch storageclass gp2 -p '{\"metadata\": {\"annotations\":{\"storageclass.kubernetes.io/is-default-class\":\"true\"}}}'"
  }
}
```

**Comment by pdecat** [comment]
This should be fixed by #325 that was released in v1.7.0: https://github.com/terraform-providers/terraform-provider-kubernetes/blob/master/CHANGELOG.md#170-may-22-2019

**Comment by barryib** [comment]
I can confirm that this solved in v1.9

**Comment by aareet** [comment]
Confirmed this works in v1.11.1
